trading_calendars/exchange_calendar_xfra.py

```diff
@@ -127,6 +128,16 @@ def regular_holidays(self):
             NewYearsEve,
         ])
 
+    @property
+    def adhoc_holidays(self):
+        return [
+            # Whit Monday observed in 2007, before it became regularly
+            # observed starting in 2015.
+            Timestamp('2007-05-28', tz='UTC'),
+            # Reformation Day was a German national holiday in 2017.
+            Timestamp('2017-10-31', tz='UTC'),
```

Comment:
For consistency with other adhocs, could define these as `WhitMonday2007`, `ReformationDay500thAnniversary` (or something to that effect) below the regular holidays in this file, and reference them here?

Reply:
Sure, I might also add `AdHoc` to the end of those names to differentiate them from the holiday calendars